modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java

```diff
@@ -1224,6 +1224,31 @@ public void testClusterSnapshotFinishedTryCancel() throws Exception {
         assertSnapshotCacheKeys(snpIg.cache(dfltCacheCfg.getName()));
     }
 
+    /** @throws Exception If fails. */
+    @Test
+    public void testClientHandlesSnapshotFailOnStartStage() throws Exception {
```

Comment:
What if assert is disabled? Should we check directory is not deleted on client if exists?